Content.Server/BarSign/Systems/BarSignSystem.cs

```diff
@@ -17,81 +17,63 @@ public class BarSignSystem : EntitySystem
 
         public override void Initialize()
         {
-            base.Initialize();
-
-            SubscribeLocalEvent<BarSignComponent, ComponentInit>(OnComponentInit);
-            SubscribeLocalEvent<BarSignComponent, MapInitEvent>(OnMapInit);
-            SubscribeLocalEvent<BarSignComponent, PowerChangedEvent>(OnPowerChanged);
-        }
-
-        public void SetBarSign(BarSignComponent component, string barSign)
-        {
-            component.CurrentSign = barSign;
-            UpdateBarSignVisuals(component);
+            SubscribeLocalEvent<BarSignComponent, PowerChangedEvent>(UpdateBarSignVisuals);
         }
 
-        private void OnComponentInit(EntityUid uid, BarSignComponent component, ComponentInit args)
+        private void UpdateBarSignVisuals(EntityUid owner, BarSignComponent component, PowerChangedEvent args)
         {
-            UpdateBarSignVisuals(component);
-        }
+            if (component.LifeStage is < ComponentLifeStage.Initialized or > ComponentLifeStage.Running) return;
 
-        private void OnMapInit(EntityUid uid, BarSignComponent component, MapInitEvent args)
-        {
-            if (component.CurrentSign != null)
+            var prototype = TryGetPrototype(component);
+            if (prototype == null)
             {
-                return;
+                prototype = Setup(owner, component);
             }
 
-            var prototypes = _prototypeManager.EnumeratePrototypes<BarSignPrototype>().Where(p => !p.Hidden)
-                .ToList();
-            var prototype = _random.Pick(prototypes);
-
-            component.CurrentSign = prototype.ID;
-        }
-
-        private void OnPowerChanged(EntityUid uid, BarSignComponent component, PowerChangedEvent args)
-        {
-            UpdateBarSignVisuals(component);
-        }
-
-        private void UpdateBarSignVisuals(BarSignComponent component)
-        {
-            if (component.CurrentSign == null)
+            var sprite = Comp<SpriteComponent>(owner);
+            if (args.Powered)
             {
-                return;
+                sprite.LayerSetState(0, prototype.Icon);
+                sprite.LayerSetShader(0, "unshaded");
             }
-
-            if (!_prototypeManager.TryIndex(component.CurrentSign, out BarSignPrototype? prototype))
+            else
             {
-                Logger.ErrorS("barSign", $"Invalid bar sign prototype: \"{component.CurrentSign}\"");
-                return;
+                sprite.LayerSetState(0, "empty");
+                sprite.LayerSetShader(0, "shaded");
             }
+        }
 
-            if (EntityManager.TryGetComponent(component.Owner, out SpriteComponent? sprite))
+        private BarSignPrototype? TryGetPrototype(BarSignComponent component)
```

Comment:
the tryget pattern usually returns a bool & uses an out param for the actual return value. if you want to stick with this signature, rename the method to GetPrototypeOrNull